src/adr/0036-against-use-of-conditional-statements-in-guice-modules.md

```diff
@@ -0,0 +1,110 @@
+hard to document and test.
+
+We thus decided to rely on Guice. Components are defined by code in a static fashion. Overriding components requires 
+explicit code modification and recompilation, thus warning the user about the impact of the choices he does, and lowering 
+the project's responsibility.
+
+With Guice we expose only supported, well tested combinations of components, thus addressing the combination issue.
+
+Instead of having a single big application being able to instantiate each and every component application, we have 
+several products defining their dependencies in a minimalistic way, relying only on the components implementation that 
+are needed.
+
+Here is the list of products we provide:
+
+ - In-Memory: A memory based James server, mainly for testing purposes
+ - Distributed James: A scalable James server, storing data in various data stores. Cassandra is used for metadata, 
+ ElasticSearch for search, RabbitMQ for messaging, and ObjectStorage for blobs.
+ - Cassandra: An implementation step toward Distributed James. It does not include messaging and ObjectStorage and should not be run in a cluster way but is still relevant for good performance.
+ - JPA: A JPA and Lucene based implementation of James. Only Derby driver is currently supported.
+ - JPA with SMTP only using derby: A minimalist SMTP server based on JPA storage technology and Derby driver
+ - JPA with SMTP only using mariadb: A minimalist SMTP server based on JPA storage technology and MariaDB driver
+
+Some components however do have several implementations a user can choose from in a given product. This is the case for:
+
+ - BlobExport: Exporting a blob from the blobStore to an external user. Two implementations are currently supported: 
+ localFiles and LinShare.
+ - Text extraction: Extracting text from attachment to enable attachment search. There is a Tika implementation, but 
+ lighter JSOUP based options are also available.
+
+In order to keep the number of products low, we decided to use conditional statements in modules based on the 
+configuration to select which one to enable at runtime. Eventually defeating the Guice adoption goals mentioned above.
+
+Finally, Blob Storing technology offers a wide combination of technologies:
+
+ - ObjectStorage in itself could implement either Swift APIs or Amazon S3 APIs
+ - We decided to keep supporting Cassandra for blob storing as an upgrade solution from Cassandra product to Distributed 
+James for existing users.
+ - Proposals such as [HybridBlobStore](0014-blobstore-storage-policies.md) and then 
+[Cassandra BlobStore cache](0025-cassandra-blob-store-cache.md) proposed to leverage Cassandra as a performance 
+(latency) enhancer for ObjectStorage technologies.
+
+Yet again it had been decided to use conditional statements in modules in order to lower the number of product.
+
+However, [Cassandra BlobStore cache](0025-cassandra-blob-store-cache.md) requires expensive resource initialization
```

Comment:
maybe you can start this paragraph with the design issue, and then explain that Cassandra cache is an example.